aten/src/ATen/native/quantized/cpu/qfc_unpack.cpp

```diff
@@ -0,0 +1,65 @@
+#include <ATen/ATen.h>
+#include <ATen/core/Type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/cpp_custom_type_hack.h>
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/quantized/Quantizer.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at {
+namespace native {
+namespace {
+
+class QFCUnpackWeightInt8 final : public c10::OperatorKernel {
+ public:
+#ifdef USE_FBGEMM
+  at::Tensor operator()(at::Tensor packed_weight) {
+    // Pull out the PackBMatrix instance from the owning tensor.
+    auto& pack_ptr = cpp_custom_type_hack::cast<PackedFCWeight>(packed_weight);
+    auto packB = pack_ptr.w.get();
```

Comment:
should we check the packed weight is actually `int8` here

Reply:
Do you mean that we need to add an ASSERT statement here? We can add that to check the type to make sure we have c10::qint8 instead of c10::quint8 for the packed_weight tensor.